lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +999,17 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive_fmt not in ['zip', 'tar']:
+        archive_fmt = 'tar'
```

Comment:
You don't need this `if` block here, since the AnsibleModule arg handling will do it for you.